Get Apps card: Ensure card is always displayed, regardless of promos or dismissal

SiobhyB opened the pull request:
Following the changes to the Get Apps card in https://github.com/Automattic/jetpack/pull/26276, there is no longer a button to dismiss the card. However, some of the surrounding code related to that preference was not deleted.

In addition, the card is displayed when [the `arePromotionsActive` function](https://github.com/Automattic/jetpack/blob/8b32bb737e90923a229a0f21cf54b7d654b59d24/projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js#L310-L319) is set to false. `arePromotionsActive` returns false in cases where a user can't purchase plans. This isn't necessary, as the apps can be used by all types of users and whether promotions are active isn't relevant to its use. 

#### Changes proposed in this Pull Request:
* All the code related to storing a user's dismissal preference is removed with this PR. This includes `appsCardDismissed()` and `dismiss_dash_app_card` mentions as well as now-unused CSS related to the dismiss button. This will prevent future confusion.
* In addition, the `arePromotionsActive` conditional has been removed from the Get Apps component, so that the card is always displayed.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:

* Navigate to _Jetpack_ → _Dashboard_ to view the card.
* Locally, set [the `arePromotionsActive` function](https://github.com/Automattic/jetpack/blob/8b32bb737e90923a229a0f21cf54b7d654b59d24/projects/plugins/jetpack/_inc/client/state/initial-state/reducer.js#L310-L319) to return `false`.
* Verify that the banner still displayed. 

#### Screenshot:

<img width="1163" alt="Screenshot 2022-10-06 at 11 57 52" src="https://user-images.githubusercontent.com/2998162/194296165-b67b9ced-50ab-4126-9382-195588dec2ae.png">

SiobhyB committed:
Remove dismissal preference for apps card

The apps card is no longer dismissable. As such, this commit remove the code that was storing a user's dismissal preference related to that card.

SiobhyB committed:
Update changelog entry

SiobhyB committed:
Display Apps card, even when promos aren't active

SiobhyB committed:
Remove now-unused CSS for dismissing card

SiobhyB committed:
Update changelog entry

This commit updates the changelog to reflect the updated scope of the PR.

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

SiobhyB committed:
Remove unused functions now that Redux isn't used

As we're no longer using states to determine whether AppsCard displays, it's no longer necessary to explore any functions along with the component. We can therefore remove the 'connect' and 'withRouter' functions.

SiobhyB committed:
Merge branch 'fix/get-apps-dismiss-option' of github.com:Automattic/Jetpack into fix/get-apps-dismiss-option

fluiddot commented on projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx:
```diff
@@ -114,8 +107,4 @@ AppsCard.propTypes = {
-	return {
-		arePromotionsActive: arePromotionsActive( state ),
-	};
-} )( withRouter( AppsCard ) );
```
We should keep wrapping `AppsCard` component with [`withRouter`](https://v5.reactrouter.com/web/api/withRouter) to have access to `location` prop, which is used in the below code line. Otherwise, the event tracked in `trackDownloadClick` is not dispatched due to an error.

https://github.com/Automattic/jetpack/blob/7d9418607207b8e111436395180fa745121ba60a/projects/plugins/jetpack/_inc/client/components/apps-card/index.jsx#L19

SiobhyB replied:
Thank you for spotting this! I've added the wrapping back in https://github.com/Automattic/jetpack/pull/26659/commits/989598271d182638a937828721d313623e34a1a2.

fluiddot reviewed (changes requested):
I tested the changes and work as expected based on the testing instructions 🎊. However, I found an issue related to tracking the download click. Once this issue is addressed, from my side the PR would be ready to merge.

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

SiobhyB committed:
Fix: Enable access to location prop in AppsCard

This commit follows up on the following feedback: https://github.com/Automattic/jetpack/pull/26659/files#r992533712

The removal of the 'withRouter' variable prevented the AppsCard component from accessing the location prop, which broke the 'trackDownloadClick' function. This commit fixes that by wrapping the component with 'withRouter' again.

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

fluiddot reviewed (approved):
LGTM 🎊 , I've just re-tested the PR and confirmed that the issue outlined in [this comment](https://github.com/Automattic/jetpack/pull/26659/files#r992533712) is addressed 🎊 .

SiobhyB committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

SiobhyB commented:
Thank you again, Carlos! ❤️ Setting as ready for the final Crew review.

coder-karen committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

coder-karen committed:
Merge branch 'trunk' into fix/get-apps-dismiss-option

coder-karen reviewed (approved):
LGTM 👍
